### PR TITLE
chore: Adds PR check that verifies migration guide when flagged as breaking change

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Adds PR check that verifies migration guide when flagged as breaking change

Link to any related issue(s): CLOUDP-241732

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Review by oarbusi — APPROVED** [review]

File: contributing/development-setup.md [hunk 1]
```diff
@@ -69,10 +69,9 @@ Use [*Conventional Commits*](https://www.conventionalcommits.org/) to name pull
 - `style: Description of the PR`: The commit updates or reformats the style of the source code, but does not otherwise change the product implementation (this correlates with Minor in Semantic Versioning).
 - `feat: Description of the PR`: a commit of the type feat introduces a new feature to the codebase (this correlates with MINOR in Semantic Versioning).
 - `deprecate: Description of the PR`: The commit deprecates existing functionality, but does not remove it from the product (this correlates with MINOR in Semantic Versioning).
-- `BREAKING CHANGE`: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.
+- `a ! in the description of the PR`: The commit introduces a breaking change (correlating with MAJOR in Semantic Versioning). A breaking change can be part of commits of any type.
```
Comment (lantoli) [inline contributing/development-setup.md:72]: this in line with the PR labeler config we use: https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/.github/pr-labeler.yml#L4

File: contributing/development-setup.md [hunk 2]
```diff
@@ -69,10 +69,9 @@ Use [*Conventional Commits*](https://www.conventionalcommits.org/) to name pull
 - `style: Description of the PR`: The commit updates or reformats the style of the source code, but does not otherwise change the product implementation (this correlates with Minor in Semantic Versioning).
```
Comment (AgustinBettati) [inline contributing/development-setup.md:69]: Was thinking if we should update our pull request template required check list with a point on the lines: If the changes include a breaking change, or a new feature that contributes to a coming major version (e.g. v1.15.0), a migration guide must be created or updated.

**Review by AgustinBettati — APPROVED** [review]
LGTM

File: contributing/development-setup.md (hunk 2 above)
Comment (lantoli) [reply contributing/development-setup.md:69]: good idea, PR template changed

**Pull request merged**